modular_skyrat/modules/modular_items/lewd_items/code/lewd_machinery/milking_machine.dm

```diff
@@ -0,0 +1,1010 @@
+
+	// // Standard behavior. Uncomment for UI debugging
+	// if(!ui)
+	// 	ui = new(user, src, "MilkingMachine", name)
+	// 	ui.open()
+	// ///////////////////////////////////////////////////////////
+
+	//Block the interface if we are in the machine. Use in production
+	if(LAZYLEN(buckled_mobs))
+		if(user != src.buckled_mobs[1])
+			if(!ui)
+				ui = new(user, src, "MilkingMachine", name)
+				ui.open()
+				return
+		else if(ui)
+			ui.close()
+			return
+	else if(!ui)
+		ui = new(user, src, "MilkingMachine", name)
+		ui.open()
+		return
+	///////////////////////////////////////
+
+// Interface data filling handler
+/obj/structure/chair/milking_machine/ui_data(mob/user)
+	var/list/data = list()
+
+	data["mobName"] = current_mob ? current_mob.name : null
+	data["mobCanLactate"] = current_breasts ? current_breasts.lactates : null
+	data["cellName"] = cell ? cell.name : null
+	data["cellMaxCharge"] = cell ? cell.maxcharge : null
+	data["cellCurrentCharge"] = cell ? cell.charge : null
+	data["beaker"] = beaker ? beaker : null
+	data["BeakerName"] = beaker ? beaker.name : null
+	data["beakerMaxVolume"] = beaker ? beaker.volume : null
+	data["beakerCurrentVolume"] = beaker ? beaker.reagents.total_volume : null
+	data["mode"] = current_mode
+	data["milkTankMaxVolume"] = max_vessel_capacity
+	data["milkTankCurrentVolume"] = milk_vessel ? milk_vessel.reagents.total_volume : null
+	data["girlcumTankMaxVolume"] = max_vessel_capacity
+	data["girlcumTankCurrentVolume"] = girlcum_vessel ? girlcum_vessel.reagents.total_volume : null
+	data["semenTankMaxVolume"] = max_vessel_capacity
+	data["semenTankCurrentVolume"] = semen_vessel ? semen_vessel.reagents.total_volume : null
+	data["current_vessel"] = current_vessel ? current_vessel : null
+	data["current_selected_organ"] = current_selected_organ ? current_selected_organ : null
+	data["current_selected_organ_name"] = current_selected_organ ? current_selected_organ.name : null
+	if(current_mob?.is_topless() || current_breasts?.visibility_preference == GENITAL_ALWAYS_SHOW)
+		data["current_breasts"] = current_breasts ? current_breasts : null
+	else
+		data["current_breasts"] = null
+
+	if(current_mob?.is_bottomless() || current_testicles?.visibility_preference == GENITAL_ALWAYS_SHOW)
+		data["current_testicles"] = current_testicles ? current_testicles : null
+	else
+		data["current_testicles"] = current_testicles = null
+
+	if(current_mob?.is_bottomless() || current_vagina?.visibility_preference == GENITAL_ALWAYS_SHOW)
+		data["current_vagina"] = current_vagina ? current_vagina : null
+	else
+		data["current_vagina"] = current_vagina = null
+
+	data["machine_color"] = machine_color
+	updateUsrDialog()
+	return data
+
+// User action handler in the interface
+/obj/structure/chair/milking_machine/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+	if(action == "ejectCreature")
+		unbuckle_mob(current_mob)
+		to_chat(usr,"<span class='notice'>You ejected creature from the machine</font>")
```

Comment:
Might want to change this so it says the name of the current mob instead of "Creature."